lib/pbench/test/unit/server/auth/test_auth.py

```diff
@@ -20,6 +20,84 @@
 from pbench.test.unit.server.conftest import jwt_secret
 
 
+@pytest.fixture
+def mock_oidc(monkeypatch, on_disk_server_config):
+    def mock_connection(
+        client_id: str, public_key: Optional[str] = None
+    ) -> PbenchServerConfig:
+        """Create a mocked Connection object whose behavior is driven off of
+        the realm name and / or client ID.
```

Comment:
This docstring is somewhat confusing, since `mock_connection()` returns a `PbenchServerConfig` object, not a `Connection` object.  The point of this function seems to be to install a mock replacement for `pbench.server.auth.Connection`...perhaps "create" isn't the best choice of word?  (This code _defines_ a mock `Connection` object and then replaces the real one with it.)

(I see that this was, as Peter would say, "a lift-and-shift"...but, it's probably worth some wordsmithing anyway, now that you've wrapped it in a fixture.)

Reply:
Getting the unit test mocks working again was a massive pain, and I didn't at the time have patience for cleaning up the comments. But, yeah, this morning it's probably worth a few cycles now that the trauma is somewhat distant. 😆